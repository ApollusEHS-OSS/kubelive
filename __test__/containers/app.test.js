'use strict';
const React = require('react');
const { shallow } = require('enzyme');
const importJsx = require('import-jsx');
const App = importJsx('../../containers/app');
const Namespaces = importJsx('../../containers/namespaces');
const Pods = importJsx('../../containers/pods');
jest.mock('../../kube/api');

describe('App', () => {
  let container;
  let namespaceContainer;

  beforeEach(() => {
    container = shallow(<App />);
    namespaceContainer = container.find(Namespaces).first();
  });

  it('should match the snapshot', () => {
    expect(container).toMatchSnapshot();
  });

  it('should set the Namespaces onNamespaceChange prop to this.onNamespaceChange', () => {
    expect(namespaceContainer.props().onNamespaceChange).toBe(
      container.instance().onNamespaceChange
    );
  });

  it('should call setState({ selectedNamespace }) when there is change in namespace', () => {
    const namespace = 'some-namespace-name';
    container.instance().onNamespaceChange(namespace);

    expect(container.state().selectedNamespace).toEqual(namespace);
  });

  it('should set the Pods namespace prop to this.state.selectedNamespace', () => {
    const namespace = 'some-namespace';

    container.setState({
      ...container.state(),
      selectedNamespace: namespace
    });
    container.update();
    const podsContainer = container.find(Pods).first();

    expect(podsContainer.props().namespace).toBe(
      container.state().selectedNamespace
    );
  });
});
