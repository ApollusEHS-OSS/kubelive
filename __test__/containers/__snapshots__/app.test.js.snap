// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<Fragment>
  <Namespaces
    onNamespaceChange={[Function]}
  />
  <Pods
    namespace=""
  />
</Fragment>
`;
